R/utils.R

```diff
@@ -30,4 +30,53 @@ is_atomic_list <- function(ll, .p) {
 
 assertthat::on_failure(is_atomic_list) <- function(call, env) {
   "Element of the list are not all of the same class"
-}
\ No newline at end of file
+}
+
+
+#'Build a `VolSurface` from an example date set
+#'
+#'This creates a [`VolSurface`][`VolSurface`] object from the example data set
+#' `volsurface.csv`.
+#'
+#' @return a `VolSurface` object using data from `volsurface.csv`
+#' @examples build_vol_surface()
+#' @export
+#' @family build object helpers
+
+build_vol_surface <- function(){
+##1.651555326
+
+  filepath <- system.file("extdata", "volsurface.csv", package = "fmbasics")
+  vols <- utils::read.csv(file = filepath, header = T )
+  spot <- 97.62
+  reference_date <- as.Date("2019-04-26")
+  tenors <- lubridate::ymd(vols$Date)
+  strikes <- seq(from = 0.05, to = 2.55, by= 0.05)*spot
+  vol_quotes <- vols[,-c(1:4) ]
+  vol_quotes <- t(vol_quotes)/100
+  vol_interp <- LinearTimeVarInterpolation()
+  vol_obj <- VolSurface(reference_date, "strike/term", vol_quotes, strikes, tenors,
```

Comment:
You don't need to pass the result to `vol_obj` and then return it on the next line. Leave the call to `VolSurface()` as the last line in the function.

Reply:
That's true, I removed the extra variable assignment. It is a cautious habit.